proxy_agent/src/service.rs

```diff
@@ -32,22 +32,40 @@ pub fn start_service(shared_state: Arc<Mutex<SharedState>>) {
         helpers::get_elapsed_time_in_millisec()
     ));
 
-    let config_start_redirector = config::get_start_redirector();
-
-    crate::key_keeper::poll_status_async(
-        Url::parse(&format!("http://{}/", constants::WIRE_SERVER_IP)).unwrap(),
-        config::get_keys_dir(),
-        config::get_poll_key_status_duration(),
-        config_start_redirector,
-        shared_state.clone(),
-    );
-
-    proxy_listener::start_async(constants::PROXY_AGENT_PORT, 20, shared_state.clone());
+    start_service_async(shared_state.clone());
 
     // TODO:: need start the monitor thread and write proxy agent status to the file
     // monitor::start_async(config::get_monitor_duration());
 }
 
+fn start_service_async(shared_state: Arc<Mutex<SharedState>>) {
+    _ = std::thread::Builder::new().spawn(move || {
```

Comment:
Instead of building a new thread directly, should we let [tokio](https://tokio.rs/tokio/tutorial/spawning) handle that?